src/trio/_tests/test_socket.py

```diff
@@ -819,7 +834,8 @@ async def test_address_in_socket_error() -> None:
         try:
             await sock.connect((address, 2))
         except OSError as e:
-            assert any(address in str(arg) for arg in e.args)
+            # the noqa is "Found assertion on exception `e` in `except` block"
+            assert any(address in str(arg) for arg in e.args)  # noqa: PT017
```

Comment:
It wants you to do this:
```python
with pytest.raises(OSError, match=...) as excinfo:
  ...
assert any(... for arg in excinfo.value.args)
```

Looks like there's a few more instances of PT017 you can fix similarly

Reply:
Changed in f89cd0a